modules/pricer/src/main/java/com/opengamma/strata/pricer/impl/rate/swap/DiscountingRatePaymentPeriodPricer.java

```diff
@@ -404,5 +444,69 @@ private void explainPresentValue(
     builder.put(ExplainKey.PAY_OFF_RATE, accrualPeriod.getNegativeRateMethod().adjust(payOffRate));
     builder.put(ExplainKey.UNIT_AMOUNT, ua);
   }
+  
+  // Sensitivity to the spread for a payment period with FLAT compounding type
+  private double pvbpCompoundedFlat(RatePaymentPeriod paymentPeriod, RatesProvider provider) {
+    int nbCmp = paymentPeriod.getAccrualPeriods().size();
+    double[] rate = new double[nbCmp];
+    int i = 0;
```

Comment:
Would prefer to see loops replaced with standard "for (int i = 0; ...)" style, rather than declaring 'i' and manually incrementing it. A stream would also work, using 'mapToDouble()' and 'toArray()'.